shared/src/androidMain/kotlin/io/customer/shared/tracking/queue/QueueTimer.kt

```diff
@@ -0,0 +1,102 @@
+package io.customer.shared.tracking.queue
+
+import android.os.CountDownTimer
+import io.customer.shared.util.Logger
+import io.customer.shared.work.JobDispatcher
+import io.customer.shared.work.JobExecutor
+import io.customer.shared.work.TimeUnit
+import io.customer.shared.work.runOnMain
+
+internal actual fun getQueueTimer(
+    logger: Logger,
+    jobExecutor: JobExecutor,
+): QueueTimer = AndroidQueueTimer(
+    logger = logger,
+    jobExecutor = jobExecutor,
+)
+
+/**
+ * [QueueTimer] implementation that uses Android [CountDownTimer] to run timer countdowns.
+ */
+private class AndroidQueueTimer(
+    private val logger: Logger,
+    override val jobExecutor: JobExecutor,
+) : QueueTimer, JobDispatcher {
+    @Volatile
+    private var countdownTimer: CountDownTimer? = null
+    private val timerAlreadyScheduled: Boolean
+        get() = countdownTimer != null
+
+    override fun onCoroutineFailed(exception: Throwable) {
+        countdownTimer = null
+        logger.error(message = "Timer failed with error ${exception.message}")
+    }
+
+    override fun schedule(force: Boolean, duration: TimeUnit.Seconds, block: () -> Unit) {
+        synchronized(this) {
+            if (!force && timerAlreadyScheduled) {
+                log("already scheduled to run. Skipping request.")
+            }
+            scheduleAndCancelPrevious(duration = duration, block = block)
+        }
+    }
+
+    private fun scheduleAndCancelPrevious(duration: TimeUnit, block: () -> Unit) {
```

Comment:
```suggestion
    private fun scheduleAndCancelPrevious(seconds: TimeUnit.Seconds, block: () -> Unit) {
```

More descriptive name of the unit used makes the code easier for me to read.

Reply:
I have made the timer class independent of units. As it relies on [`toMillis`](https://github.com/customerio/mobile-shared/blob/945c9cb4ca5e02374ef5f3a305b254dbc337a58a/shared/src/androidMain/kotlin/io/customer/shared/tracking/queue/QueueTimer.kt#L61) from base class and makes it easier to pass seconds, milliseconds or any other derived class from [`TimeUnit`](https://github.com/customerio/mobile-shared/blob/945c9cb4ca5e02374ef5f3a305b254dbc337a58a/shared/src/commonMain/kotlin/io/customer/shared/work/TimeUnit.kt). The [caller](https://github.com/customerio/mobile-shared/blob/945c9cb4ca5e02374ef5f3a305b254dbc337a58a/shared/src/commonMain/kotlin/io/customer/shared/tracking/queue/QueueDispatcher.kt#L106) however, makes it clear which unit is being passed.